reference/api/documents.mdx

```diff
@@ -248,6 +248,45 @@ An array of documents. Each document is represented as a JSON object.
 ]
 ```
 
+#### `_vectors`
+
+`_vectors` is a special document attribute containing vector embeddings for AI-powered search. It accepts different types of values.
+
+`_vectors` may be an array of numbers:
+
+```json
+[
+  {
+    "id": 452,
+    "title": "Female Trouble",
+    "overview": "Delinquent high school student Dawn Davenport runs away from home and embarks upon a life of crime.",
+    "_vectors": [0.0, 0.1, 0.003, 0.75]
+  },
+]
+```
+
+`_vectors` may also be an object with two fields, `embeddings` and `regenerate`:
+
+```json
+[
+  {
+    "id": 452,
+    "title": "Female Trouble",
+    "overview": "Delinquent high school student Dawn Davenport runs away from home and embarks upon a life of crime.",
+    "_vectors": {
+      "embeddings": [],
+      "regenerate": true
+    }
+  }
+]
+```
+
+`embeddings` must be an array of numbers.
```

Comment:
```suggestion
If provided, `embeddings` must be an array of numbers, representing a single embedding for that embedder for that document, or an array of arrays of numbers, representing multiple embeddings for that embedder for that document.

If missing or `null`, the embeddings for that document will remain at their current value, except if `regenerate` is `true` and other fields in the document were modified in the same document update.
```

I suggest we introduce `regenerate` first.

Reply:
Could you expand on this? I find the explanation flows better by saying "embeddings defines the vectors for this document" -> "regenerate regenerates the embeddings your previously defined".

Reply:
Sure.

- `embeddings` overrides the current embeddings of the documents
- separately from this, `regenerate` describes the behavior when the document is updated.

So, the flow could be:

By default, Meilisearch will generate embeddings when a document is added and each time the document is updated such that the filling the `documentTemplate` with the document's data produces a different text as before the update.

When setting `regenerate` to `false,` this behavior will be disabled, and the embedding will remain the same regardless of the document's updates.

Additionally, if providing the `embeddings` field, the embeddings for that document and embedder will be overridden. 

The various combinations of `regenerate` and `embeddings` can achieve various use cases:

### 1. Providing embeddings to a `userProvided` embedder

Setting `regenerate` to `false` and the `embeddings` to the target value will achieve this use case.

### 2. Importing embeddings that were created externally using a similar embedder once, then letting regeneration occurs when updating documents.

Import the embeddings in `embeddings`, and set `regenerate` to `true` so that future updates regenerate the embeddings.

### 3. Opting-out of embeddings for a single document.

Similar to use case (1), with `regenerate` set to `false`, but with an `embeddings` equal to `[]`

### 4. "Freezing" the current values of embeddings.

Pass only `regenerate` to `false` without specifying `embeddings`. The document will keep the embeddings currently in its database, regardless of future updates.

Reply:
I'm still digesting this, but a question. Are these two examples equivalent?

```json
{
  "default": [0.1, 0.2, 0.3]
}
```

```json
{
  "default": {
    "embeddings": [0.1, 0.2, 0.3],
    "regenerate" : true
}
```

If they're identical, what is the use case for the simple array format? Wouldn't it be better to remove it and simplify the API, so users don't need to think which format is best for them?

Reply:
Ah, ok, so by default Meilisearch will not regenerate custom embeddings. That makes sense, thanks.

I think you have found the correct framing for this. The new format (with embeddings and regenerate) is the actual format, whilst the the regular array is simply a shorthand. Idk, perhaps it was obvious to you, but I kept seeing them as two separate things.